shared/src/utils/validators/is-value.ts

```diff
@@ -0,0 +1,25 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { isIn } from 'class-validator';
+import type { GenericPropertyDecorator } from './generic-property-decorator';
+import { makeValidator } from './make-validator';
+
+export function isValue<T>(
+    expectedValue: T,
+    valueToBeValidated: unknown
+): boolean {
+    return isIn(valueToBeValidated, [expectedValue]);
+}
+
+// With TS 5.0 use `const T` (https://github.com/microsoft/TypeScript/pull/51865)
```

Comment:
```suggestion
// TODO [typescript@>=5.0]: Use `const T` (https://github.com/microsoft/TypeScript/pull/51865)
```

See https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md

Reply:
FYI: Maybe we should add this to the documentation. I'm considering refactoring it a bit and adding a `DEVELOPMENT.md` to the root...